modules/apps/frontend-js/frontend-js-test/src/testFunctional/tests/TranslationManager.testcase

```diff
@@ -40,6 +40,64 @@ definition {
 		}
 	}
 
+	@description = "LPS-148144. Validates language status after re-add language on manage translations modal."
+	@priority = "3"
+	test AssertLanguageStatusOnModal {
+		task ("Given manage translations modal") {
+			TranslationManagerPortlet.goToManageTranslationsModal(
+				key_currentLocale = "en-us",
+				key_title = "Admin");
+		}
+
+		task ("When delete Catalan language on Manage Translations modal") {
+			ManageTranslations.deleteLanguage(key_language = "Catalan");
+		}
+
+		task ("Then re-add the Catalan language on Manage Translations modal") {
+			ManageTranslations.addLanguage(key_locale = "ca-ES");
+		}
+
+		task ("Assert the Catalan language is still translated status") {
+			AssertElementPresent(
+				key_language = "Catalan",
+				key_text = "Translated",
+				locator1 = "ManageTranslations#LANGUAGE_STATUS");
+
+			Click(locator1 = "Modal#CLOSE_BUTTON");
+		}
+
+		task ("Given manage translations modal") {
+			TranslationManagerPortlet.goToManageTranslationsModal(
+				key_currentLocale = "en-us",
+				key_title = "Admin");
+		}
+
+		task ("When save delete Catalan language on Manage Translations modal") {
+			ManageTranslations.deleteLanguage(key_language = "French");
+
+			Click(locator1 = "Button#DONE");
+		}
+
+		task ("Given manage translations modal") {
+			TranslationManagerPortlet.goToManageTranslationsModal(
+				key_currentLocale = "en-us",
+				key_title = "Admin");
+		}
+
+		task ("Then re-add the French language on Manage Translations modal") {
+			ManageTranslations.addLanguage(key_locale = "fr-FR");
+		}
+
+		task ("Then assert the Catalan language is changed to Not Translated status") {
+			AssertElementPresent(
+				key_language = "French",
+				key_text = "Not Translated",
+				locator1 = "ManageTranslations#LANGUAGE_STATUS");
+
+			Click(locator1 = "Modal#CLOSE_BUTTON");
+		}
+	}
+
```

Comment:
We need to keep testcases separate and not keep combining them like this. If we have to do two or more Given when then sequences then this indicates two separate tests and is not good practice to combine. When a QA/dev reads the test report we want to know exactly what failed as quickly as possible and not have to decide which of the Given when then situations caused the issue.